lib/db/mem.js

```diff
@@ -613,14 +613,27 @@ module.exports = function (log, error) {
 
     var accountId = sessionTokens[id].uid.toString('hex')
     var account = accounts[accountId]
-    item.emailVerified = account.emailVerified
-    item.email = account.email
-    item.emailCode = account.emailCode
-    item.verifierSetAt = account.verifierSetAt
-    item.locale = account.locale
-    item.accountCreatedAt = account.createdAt
 
-    return P.resolve(item)
+    return this.accountEmails(accountId)
+      .then((emails) => {
+
+        // Set the primary email on the sessionToken, which
+        // could be different from the email on the account object
+        emails.some((email) => {
+          if (email.isPrimary) {
+            item.emailVerified = email.isVerified
+            item.email = email.email
+            item.emailCode = email.emailCode
+            return true
+          }
+        })
+
+        item.verifierSetAt = account.verifierSetAt
+        item.locale = account.locale
+        item.accountCreatedAt = account.createdAt
```

Comment:
It feels to me like these 3 should logically be set immediately after `var account = accounts[accountId]` on line 615.